Sources/Fluent/Entity/EntityError.swift

```diff
@@ -9,27 +9,77 @@ public enum EntityError: Error {
     case unspecified(Error)
+}
+
+extension EntityError: Debuggable {
+    public var identifier: String {
+        switch self {
+        case .noDatabase:
+            return "noDatabase"
+        case .noId:
+            return "noId"
+        case .doesntExist:
+            return "doesntExist"
+        case .unspecified:
+            return "unspecified"
+        }
+    }
+
+    public var reason: String {
+        switch self {
+        case .noDatabase:
+            return "missing database"
+        case .noId:
+            return "missing id, entities can't exist in a fluent database without their id being set."
+        case .doesntExist:
+            return "doesn't exist yet"
+        case .unspecified(let error):
+            return "unspecified \(error)"
+        }
+    }
 
-    public var description: String {
-        let entity: Entity.Type?
-        let reason: String
+    public var possibleCauses: [String] {
         switch self {
         case .noDatabase(let e):
-            entity = e
-            reason = "missing database, make sure to call `database.prepare(\(e).self)` or ensure that it's added to your Droplet's `preparations` with `drop.preprations.append(\(e).self)"
+            return [
+                "\(e) hasn't been added to Droplet's preparations",
+                "\(e) is being used manually, and database hasn't been set"
+            ]
         case .noId(let e):
-            entity = e
-            reason = "missing id, entities can't exist in a fluent database without their id being set. Make sure you're fetching properly in fluent or setting this manually if necessary."
+            return [
+                "\(e) not fetched properly",
+                "loaded manually, forgot to set id"
+            ]
         case .doesntExist(let e):
-            entity = e
-            reason = "this object wasn't fetched from the database properly. If you're using custom behavior, make sure to set exists to true after fetching from database"
-        case .unspecified(let err):
-            entity = nil
-            reason = "extension error found - \(err)"
+            return [
+                "\(e) not fetched properly",
+                "loaded manually, forgot to set exists"
+            ]
+        case .unspecified(_):
+            return [
+                "received unspecified or unknown error"
+            ]
         }
+    }
 
-        let type = entity.flatMap { "- \($0) -" } ?? "-"
-        return "\(EntityError.self) \(type) \(reason)"
+    public var suggestedFixes: [String] {
+        switch self {
+        case .noDatabase(let e):
+            return [
+                "make sure to call `database.prepare(\(e).self)` or ensure that it's added to your Droplet's `preparations` with `drop.preprations.append(\(e).self)"
+            ]
+        case .noId(_):
+            return [
+                "make sure you're fetching properly from fluent or setting 'exists' manually if necessary."
```

Comment:
should we avoid adding `.` to any of these for consistency? it would probably be easier to add a period as we print these, or possibly add commas, semicolons, etc when joining them together.